server/expression/executor_test.go

```diff
@@ -373,6 +373,69 @@ func TestArrayExecution(t *testing.T) {
 			Query:      `[31,35,39] contains 42`,
 			ShouldPass: false,
 		},
+		{
+			Name:       "should_identify_array_instead_of_json",
+			Query:      `["{}", "{}", "{}"] | type = "array"`,
+			ShouldPass: true,
+		},
+	}
+
+	executeTestCases(t, testCases)
+}
+
+func TestJSONExecution(t *testing.T) {
+	testCases := []executorTestCase{
+		{
+			Name:       "should_identify_json_input",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com"}' | type = "json"`,
+			ShouldPass: true,
+		},
+		{
+			Name:       "should_be_able_to_compare_with_subset",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com"}' contains '{"email": "john@company.com", "name": "john"}'`,
+			ShouldPass: true,
+		},
+		{
+			Name:       "should_be_able_to_compare_with_subset_ignoring_order",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com"}' contains '{"email": "john@company.com", "name": "john", "age": 32}'`,
+			ShouldPass: true,
+		},
+		{
+			Name:       "should_be_able_to_compare_deep_objects_in_subset",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com", "company": {"name": "Company", "address": "1234 Agora Street"}}' contains '{"email": "john@company.com", "name": "john", "company": {"name": "Company"}}'`,
+			ShouldPass: true,
```

Comment:
Maybe a test with deep objects and arrays?
```
{"name": "john", "age": 32, "email": "john@company.com", "company": {"name": "Company", "address": "1234 Agora Street", "telephones": ["01", "02", "03"]}}' contains '{"email": "john@company.com", "name": "john", "company": {"name": "Company", "telephones": ["01", "02", "03"]}}
```

Reply:
Added